### PR TITLE
LedgerStorage: Only rollback after begin()

### PR DESCRIPTION
**Pull request by Geod24**
A common mistake which was also done in Agora.

### CODE REVIEW

File: src/modules/storage/LedgerStorage.ts [hunk 1]
```diff
@@ -164,11 +164,11 @@ export class LedgerStorage extends Storages
         return new Promise<void>((resolve, reject) =>
         {
             (async () => {
+                let block: Block = new Block();
+                block.parseJSON(data);
+                await this.begin();
```
Comment (MichaelKim20) [inline src/modules/storage/LedgerStorage.ts:169]: `block.parseJSON(data);` and `await this.begin()` 
Exception may occur here as well.
`reject()` should be called when an exception occurs here.

Comment (TrustHenry) [reply src/modules/storage/LedgerStorage.ts:169]: Yes, I think so, too.
I think `let block: Block = new Block(); block.parseJSON(data);` can be move in `try`.

Comment (MichaelKim20) [reply src/modules/storage/LedgerStorage.ts:169]: I have confirmed that when an exception occurs in the code below, it is not handled normally.
``` TypeScript
        function testThrowError (value: number): Promise<number>
        {
            return new Promise<number>((resolve, reject) => {
                if (value === 0)
                    throw new Error("error");

                resolve(value*value);
            })
        }

        try
        {
            (async () =>
            {
                await testThrowError(0);
            })();
        }
        catch (e)
        {
            console.log(e);
        }
```

When I changed it as follows, I checked that it was handled normally.
``` TypeScript
        (async () =>
        {
            try
            {
                await testThrowError(0);
            }
            catch (e)
            {
                console.log(e);
            }
        })();
```

**Comment by Geod24** [comment]
Replaced by #163

**Pull request closed without merging**